Optifine assumes the server port is always 25565 on Java 9+

Using vanilla Minecraft 1.14.4 with Java 13
--
In this example I'll be connecting to:

![Untitled](https://user-images.githubusercontent.com/10485646/65708739-3079b480-e087-11e9-95fe-d81e5f8d7b58.png)


I am able to ping the server:

![Untitled](https://user-images.githubusercontent.com/10485646/65708000-86e5f380-e085-11e9-8aea-f588cec6d087.png)

Here's what the logs show:
```
[17:52:29] [Client thread/INFO]: Connecting to phan024.ddns.net., 25565
[17:52:29] [Client thread/WARN]: Unknown custom packed identifier: minecraft:register
[17:52:32] [Client thread/WARN]: Unknown custom packed identifier: minecraft:register
[17:52:32] [Client thread/INFO]: [CHAT]                                      
[17:52:32] [Client thread/INFO]: Loaded 7 advancements
```

Everything works correctly when using default Minecraft but there are problems when I start using Optifine.

OptiFine 1.14.4 HD U F4 pre5 with Java 13
--
I am connecting to the same server however I am unable to ping the server or connect to it:

![Untitled](https://user-images.githubusercontent.com/10485646/65707586-c233f280-e084-11e9-9053-75a2c081b467.png)

Here's what the logs show:
```
[17:42:13] [main/INFO]: Connecting to play.phanaticmc.com, 25565
[17:42:13] [Server Connector #1/ERROR]: Couldn't connect to server
java.net.SocketException: Network is unreachable
```

The issue is that Optifine with Java 9 and above for some reason will use the IP address you have specified and will disregard any SRV records the server has set in place. The log shows that when I'm using Optifine with Java 13 it is trying to connect me to `play.phanaticmc.com, 25565` however when using default Minecraft 1.14.4 with Java 13 it is pinging the server correctly and allows me to connect, the log shows that it is taking me to `phan024.ddns.net., 25565`. For some reason Optifine is not doing DNS lookups while regular Minecraft does do DNS lookups. I'm aware that Optifine does not intefere with network related things, but I believe this issue IS related to Optifine because it only happens when I'm using Optifine and the logs prove that Optifine is not behaving the same as regular Minecraft when attempting to connect to servers that use SRV records.

The DNS SRV resolving is done via reflection and Java 9 restricts reflection calls between modules.
Vanilla is running from the classpath while OptiFine is being loaded via LaunchWrapper. Apparently Java 9 sets different reflection restrictions depending from where the code is loaded.
The exception is:
```
javax.naming.NoInitialContextException: Cannot instantiate class: com.sun.jndi.dns.DnsContextFactory
	at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:719) ~[?:?]
	at javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305) ~[?:?]
	at javax.naming.InitialContext.init(InitialContext.java:236) ~[?:?]
	at javax.naming.InitialContext.<init>(InitialContext.java:208) ~[?:?]
	at javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:101) ~[?:?]
	at dkh.b(ServerAddress.java:107) ~[dkh.class:?]
	at dkh.a(ServerAddress.java:80) ~[dkh.class:?]
	at dkk.a(SourceFile:52) ~[dkk.class:?]
	at der$d.d(SourceFile:233) ~[der$d.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:299) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) [?:?]
	at java.lang.Thread.run(Thread.java:844) [?:?]
Caused by: java.lang.ClassNotFoundException: com.sun.jndi.dns.DnsContextFactory
	at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:185) ~[launchwrapper-of-2.1.jar:2.1]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:563) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:496) ~[?:?]
	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:375) ~[?:?]
	at com.sun.naming.internal.VersionHelper.loadClass(VersionHelper.java:107) ~[?:?]
	at com.sun.naming.internal.VersionHelper.loadClass(VersionHelper.java:82) ~[?:?]
	at javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:716) ~[?:?]
	... 14 more
```

With Launchwrapper bugfix to show the real problem:
```
[OptiFine] getServerAddress
javax.naming.NoInitialContextException: Cannot instantiate class: com.sun.jndi.dns.DnsContextFactory [Root exception is java.lang.IllegalAccessException: class javax.naming.spi.NamingManager (in module java.naming) cannot access class com.sun.jndi.dns.DnsContextFactory (in module jdk.naming.dns) because module jdk.naming.dns does not export com.sun.jndi.dns to module java.naming]
	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:719)
	at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at java.naming/javax.naming.InitialContext.init(InitialContext.java:236)
	at java.naming/javax.naming.InitialContext.<init>(InitialContext.java:208)
	at java.naming/javax.naming.directory.InitialDirContext.<init>(InitialDirContext.java:130)
	at dkh.b(ServerAddress.java:107)
	at dkh.a(ServerAddress.java:80)
	at dkk.a(SourceFile:52)
	at der$d.d(SourceFile:233)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:835)
Caused by: java.lang.IllegalAccessException: class javax.naming.spi.NamingManager (in module java.naming) cannot access class com.sun.jndi.dns.DnsContextFactory (in module jdk.naming.dns) because module jdk.naming.dns does not export com.sun.jndi.dns to module java.naming
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:376)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:639)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:490)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:166)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:404)
	at java.base/java.lang.Class.newInstance(Class.java:590)
	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:716)
	... 14 more
```

--- a/src/main/java/net/minecraft/launchwrapper/LaunchClassLoader.java
+++ b/src/main/java/net/minecraft/launchwrapper/LaunchClassLoader.java
@@ -44,7 +44,7 @@ public class LaunchClassLoader extends URLClassLoader {
     private static File tempFolder = null;
 
     public LaunchClassLoader(URL[] sources) {
-        super(sources, null);
+        super(sources, getParentClassLoader());
         this.sources = new ArrayList<URL>(Arrays.asList(sources));
 
         // classloader exclusions
@@ -384,5 +384,16 @@ public class LaunchClassLoader extends URLClassLoader {
 
     public void clearNegativeEntries(Set<String> entriesToClear) {
         negativeResourceCache.removeAll(entriesToClear);
+    }
+
+    private static ClassLoader getParentClassLoader() {
+        if (!System.getProperty("java.version").startsWith("1.")) {
+            try {
+                return (ClassLoader) ClassLoader.class.getDeclaredMethod("getPlatformClassLoader").invoke(null);
+            } catch (Throwable t) {
+                LogWrapper.warning("No platform classloader: " + System.getProperty("java.version"));
+            }
+        }
+        return null;
     }
 }

